datalad/config.py

```diff
@@ -54,6 +55,7 @@
 
 # we cannot import external_versions here, as the cfg comes before anything
 # and we would have circular imports
+@memoize
 def get_git_version(runner):
```

Comment:
Hmm, I wonder why config.py should be the home of `get_git_version`, which is then used by external_versions.py, rather than external_versions.py defining `get_git_version`.  Then config.py could go through the standard `external_versions` interface, and the caching the `git version` result would already be taken care of.

Reply:
why? hm, per comment above the function that we cannot import `external_versions` here.
or am I missing smth.
Related: as I have noted (in the commit msg?) this version check is primarily to check against now elderly git 2.8.  As of 7.20191218~26  git-annex debian pkg demands git >= 2.22 (to avoid memory leaks), but in principle it is ok with git 2.1. So it seems that we are doomed to carry this check for just a bit longer.

Reply:
> why? hm, per comment above the function that we cannot import `external_versions` here.
> or am I missing smth

No, I am.  I somehow managed to not see the comment in the context lines of the diff.  Sorry for the sloppy review.